### DISCUSSION TITLE
Floating-point from Real numbers

### DISCUSSION THREAD

**Discussion by baierd** [cvc5/cvc5#8219] [Q&A] [answered]
Hello,
in CVC4 one could create Floating-points from Real numbers, but it seems like this was removed and you can only create them using bitvectors now. Would it be possible to re-add the Real to FP conversion?

Thank you for your work and your help!
Best,
Daniel

**Comment by martin-cs** [comment]
Hi Baier,

When you say "In CVC4" are you talking about the C++ API, CVC language, SMT-LIB input or something else?  The CVC language has been dropped and conversions to and from Real/FP should work in the SMT-LIB input.

**Reply by baierd** [reply]
Sorry, i meant the C++ API of CVC4 (to be more exact i meant the Java API that existed a while ago, but since thats based on the C++ API it doesn't matter).
I work over at [JavaSMT](https://github.com/sosy-lab/java-smt) and we try to seperate the API from SMTLIB2 parsing as much as possible. And since it seems to be already implemented in the parser, could you add an API call for that conversion as well? (We use the Java bindings btw.)

Best,
Daniel

**Comment by martin-cs** [comment]
@mudathirmahgoub Please can you look at this?

**Comment by aniemetz** [comment, marked as answer]
Hi @baierd , we still have means to convert a Real term to a FloatingPoint term via `mkOp` of kind `FLOATINGPOINT_TO_FP_REAL`:
https://github.com/cvc5/cvc5/blob/master/src/api/java/io/github/cvc5/api/Solver.java#L746
https://github.com/cvc5/cvc5/blob/master/src/api/cpp/cvc5_kind.h#L1695

Is that what you were looking for?

**Comment by baierd** [comment]
Hi @aniemetz,
yes, now i get it, the naming was confusing. I am sorry.

Thank you both for your help!

**Reply by aniemetz** [reply]
I agree, the naming is confusing. As a heads-up, we attempt to address this here: https://github.com/cvc5/cvc5/pull/8285

**Reply by baierd** [reply]
Thank you for that! Also, the descriptions need some updates. For example `FLOATINGPOINT_TO_FP_REAL` in https://github.com/cvc5/cvc5/blob/master/src/api/cpp/cvc5_kind.h#L1695 asks for a Term of type FloatingPoint and the rounding term is missing from the `mkTerm()` below.

**Reply by aniemetz** [reply]
Good catch! Adressed here: https://github.com/cvc5/cvc5/pull/8329
